### PR TITLE
Enable logging hparams only if there are any

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Partly resolves: #11103

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @awaelchli @edward-io @borda @ananthsub

### CODE REVIEW

**Comment by carmocca** [comment]
The changes look good, but I'm wondering if there was a good reason to set True by default. Perhaps for sanity checking? to make sure logging works at all?

Did you try git blaming the addition and see if there was any discussion about it?

File: pytorch_lightning/core/mixins/hparams_mixin.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ class HyperparametersMixin:
 
     def __init__(self) -> None:
         super().__init__()
-        self._log_hyperparams = True
+        self._log_hyperparams = False
```
Comment (awaelchli) [inline pytorch_lightning/core/mixins/hparams_mixin.py:31]: Can you remind us where do we check whether we need to log or not? I.e., given this fix, there must be a check somewhere whether params were logged or not.

**Review by awaelchli — APPROVED** [review]
As far as I can see this is only getting set when calling `self.save_hyperparameters()` so this should be good!

File: pytorch_lightning/core/mixins/hparams_mixin.py (hunk 1 above)
Comment (rohitgr7) [reply pytorch_lightning/core/mixins/hparams_mixin.py:31]: here: https://github.com/PyTorchLightning/pytorch-lightning/blob/46d6fbf11b7c1d04edefa121675751db8ca02b18/pytorch_lightning/trainer/trainer.py#L1193-L1203

plus I have updated a test for it.

**Review by ananthsub — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Pull request merged**